// @enxoval/quality/dist/src/eslint/base.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.base = base;
const parser_1 = __importDefault(require("@typescript-eslint/parser"));
const eslint_plugin_1 = __importDefault(require("@typescript-eslint/eslint-plugin"));
const eslint_config_prettier_1 = __importDefault(require("eslint-config-prettier"));
const FRAMEWORK_IMPORTS = [
    { name: 'typeorm', message: 'Use lib/db instead.' },
    { name: 'kafkajs', message: 'Use lib/messaging instead.' },
    { name: 'fastify', message: 'Use lib/http instead.' },
    { name: 'pino', message: 'Use lib/observability instead.' },
    { name: 'vitest', message: 'Use lib/testing instead.' },
];
const DOMAIN_LAYERS = [
    'src/adapters/**',
    'src/controllers/**',
    'src/logic/**',
    'src/model/**',
    'src/wire/**',
];
function base() {
    return [
        {
            files: ['src/**/*.ts'],
            languageOptions: {
                parser: parser_1.default,
                parserOptions: { projectService: true },
            },
            plugins: { '@typescript-eslint': eslint_plugin_1.default },
            rules: {
                'no-restricted-imports': ['error', { paths: FRAMEWORK_IMPORTS }],
            },
        },
        {
            files: DOMAIN_LAYERS,
            rules: {
                'no-restricted-syntax': [
                    'error',
                    {
                        selector: 'FunctionDeclaration',
                        message: 'Use fn() or asyncFn() from lib/types instead of function declarations.',
                    },
                    {
                        selector: "VariableDeclarator[init.type='ArrowFunctionExpression']",
                        message: 'Use fn() or asyncFn() from lib/types instead of arrow function variables.',
                    },
                    {
                        selector: "VariableDeclarator[init.type='FunctionExpression']",
                        message: 'Use fn() or asyncFn() from lib/types instead of function expressions.',
                    },
                ],
            },
        },
        eslint_config_prettier_1.default,
    ];
}
